lib/facility.rb

```diff
@@ -1,14 +1,54 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil if !services.include?('Vehicle Registration')
+    @registered_vehicles << vehicle
+    if vehicle.plate_type == :antique
+      @collected_fees += 25
+    elsif vehicle.plate_type == :ev
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end
+    @collected_fees
+  end
+
+  def administer_written_test(registrant)
+    return false unless services.include?('Written Test') && registrant.permit? && registrant.age >= 16
+    registrant.license_data[:written] = true
+    true
+  end
+
+  def administer_road_test(registrant)
+    return false unless services.include?('Road Test') && registrant.license_data[:written]
+    registrant.license_data[:license] = true
+    true
+  end
+
+  def renew_drivers_license(registrant)
+    return false unless services.include?('Renew License')
+    return false unless registrant.license_data[:written] && registrant.license_data[:license]
```

Comment:
Breaking all these cases up into different lines was a good call for readability!